Runtime/Behaviours/CinemachineVirtualCamera.cs

```diff
@@ -493,27 +493,34 @@ private CameraState CalculateNewState(Vector3 worldUp, float deltaTime)
                 for (int i = 0; i < m_ComponentPipeline.Length; ++i)
                     m_ComponentPipeline[i].PrePipelineMutateCameraState(ref state, deltaTime);
 
-                CinemachineComponentBase postAimBody = null;
                 int componentIndex = 0;
+                CinemachineComponentBase postAimBody = null;
                 for (var stage = CinemachineCore.Stage.Body; stage <= CinemachineCore.Stage.Finalize; ++stage)
                 {
-                    if (stage == CinemachineCore.Stage.Finalize && postAimBody != null)
-                        postAimBody.MutateCameraState(ref state, deltaTime);
-
                     var c = componentIndex < m_ComponentPipeline.Length 
                         ? m_ComponentPipeline[componentIndex] : null;
                     if (c != null && stage == c.Stage)
                     {
                         ++componentIndex;
                         if (stage == CinemachineCore.Stage.Body && c.BodyAppliesAfterAim)
+                        {
                             postAimBody = c;
-                        else
-                            c.MutateCameraState(ref state, deltaTime);
+                            continue; // do the body stage of the pipeline later
+                        }
+                        c.MutateCameraState(ref state, deltaTime);
                     }
-                    else if (stage == CinemachineCore.Stage.Aim)
-                        state.BlendHint |= CameraState.BlendHintValue.IgnoreLookAtTarget;
-
                     InvokePostPipelineStageCallback(this, stage, ref state, deltaTime);
+
+                    if (stage == CinemachineCore.Stage.Aim)
+                    {
+                        if (c == null)
+                            state.BlendHint |= CameraState.BlendHintValue.IgnoreLookAtTarget;
+                        if (postAimBody != null)
+                        {
+                            postAimBody.MutateCameraState(ref state, deltaTime);
+                            InvokePostPipelineStageCallback(this, CinemachineCore.Stage.Body, ref state, deltaTime);
+                        }
+                    }
```

Comment:
This part is a bit difficult to understand at first look. It looks like _InvokePostPipelineStageCallback_ is called twice if stage is CinemachineCore.Stage.Aim. But postAimBody is null in that case, so it's ok. :)

Reply:
What is supposed to happen is that if there is a Body component with the BodyAppliesAfterAim property, the pipeline order is reversed: Aim then Body instead of Body then Aim.  Does that make sense?